libs/sdk-ui-dashboard/src/presentation/scheduledEmail/DefaultScheduledEmailDialog/utils/repeat.ts

```diff
@@ -151,3 +150,82 @@ export function setWeeklyRepeat(repeatData: IScheduleEmailRepeat, startDate: Dat
         },
     };
 }
+
+const MONTH_INDEX = 1;
+const WEEK_INDEX = 2;
+const DAY_INDEX = 3;
+const HOUR_INDEX = 4;
+const MINUTE_INDEX = 5;
+const SECOND_INDEX = 6;
+
+function getFirstFragment(fragments: number[]): { firstFragmentIndex: number; firstFragmentValue: number } {
+    for (let index = 0; index < fragments.length; index++) {
+        const fragment = fragments[index];
+        if (fragment !== 0) {
+            return {
+                firstFragmentIndex: index,
+                firstFragmentValue: fragment || 0,
+            };
+        }
+    }
```

Comment:
How about using `findIndex` and returning it together with `fragments[index]`?

Reply:
👍 done